tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -3116,7 +3163,23 @@ ReductionCodegenInfo IrEmitterUnnested::ComputeReductionCodegenInfo(
       !IsUnrollingColumnReductionBeneficial(unnested_hlo, input_shape,
                                             reduction_dimensions.dimensions[2]);
 
-  if (!dilated_x && !reduction_dimensions.is_row_reduction) {
+  KernelMappingScheme::IndexingOrder indexing_order;
```

Comment:
Use the lambda trick:
```
IndexingOrder indexing_order = [&] {
  ...
  return kLinearStridedIndexingX;
}();
```

Reply:
I did. Normally I use it for const variable with complex initialization. What gain do you see here?
I modify that variable later in some condition.